graphblas/tests/test_matrix.py

```diff
@@ -4393,3 +4395,104 @@ def test_power(A):
     B = A[:2, :3].new()
     with pytest.raises(DimensionMismatch):
         B.power(2)
+
+
+def test_setdiag():
+    A = Matrix(int, 2, 3)
+    A.setdiag(1)
+    expected = Matrix(int, 2, 3)
+    expected[0, 0] = 1
+    expected[1, 1] = 1
+    assert A.isequal(expected)
+    A.setdiag(Scalar.from_value(2), 2)
+    expected[0, 2] = 2
+    assert A.isequal(expected)
+    A.setdiag(3, k=-1)
+    expected[1, 0] = 3
+    assert A.isequal(expected)
+    # List (or array) is treated as dense
+    A.setdiag([10, 20], 1)
+    expected[0, 1] = 10
+    expected[1, 2] = 20
+    assert A.isequal(expected)
+    # size 0 diagonals (doesn't set anything, but valid call)
+    A.setdiag(-1, 3)
+    assert A.isequal(expected)
+    A.setdiag(-1, -2)
+    assert A.isequal(expected)
```

Comment:
I'm confused why these are valid calls. A size 0 diagonal feels like a diagonal out of range.

Reply:
Thanks for paying attention ;)

I flipped a coin for this one, 'cause it could be valid-ish to assign a length 0 vector to a length 0 diagonal, right?

Anyway, I changed the limits to raise for these cases.

Reply:
For a 0-length vector where k == 0, I could maybe see an exception. But if k != 0 and the off-diagonal is length zero, that is out of range. I think raising is the correct approach.